lib/fs.js

```diff
@@ -2008,13 +2008,13 @@ function ReadStream(path, options) {
   this.closed = false;
 
   if (this.start !== undefined) {
-    if (typeof this.start !== 'number') {
-      throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
+    if (!Number.isInteger(this.start)) {
+      throw new ERR_INVALID_ARG_TYPE('start', 'integer', this.start);
```

Comment:
It would be great to use much more detailed errors here. As @anliting pointed out it is not always correct to throw a `ERR_INVALID_ARG_TYPE` here.

What you can do is to check here what exactly you got that does not fit the expectations as in:

```js
if (typeof this.start !== 'number') {
  throw new ERR_INVALID_ARG_TYPE('options.start', 'number', options.start);
}
throw new ERR_OUT_OF_RANGE('options.start', 'an integer', options.start);
```

The same applies to all other errors in here. Please also be aware how I changed `start` to `options.start` (not 'this.start' as that is not the input).

Reply:
@BridgeAR Yes, doing things right is more important than having program run faster just a little bit; sorry for suggesting "just choose `ERR_OUT_OF_RANGE`".

Reply:
Your suggestion is totally fine. It will only be slower in the error case (when `Number.isInteger returns false).